cli/cmds/policy_list.go

```diff
@@ -0,0 +1,129 @@
+package cmds
+
+import (
+	"context"
+
+	"github.com/rancher/k3k/pkg/apis/k3k.io/v1alpha1"
+	"github.com/urfave/cli/v2"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/client-go/util/jsonpath"
+)
+
+func NewPolicyListCmd(appCtx *AppContext) *cli.Command {
+	return &cli.Command{
+		Name:            "list",
+		Usage:           "List all the existing policies",
+		UsageText:       "k3kcli policy list [command options]",
+		Action:          policyList(appCtx),
+		Flags:           WithCommonFlags(appCtx),
+		HideHelpCommand: true,
+	}
+}
+
+func policyList(appCtx *AppContext) cli.ActionFunc {
+	return func(clx *cli.Context) error {
+		ctx := context.Background()
+		client := appCtx.Client
+
+		if clx.NArg() > 0 {
+			return cli.ShowSubcommandHelp(clx)
+		}
+
+		var policies v1alpha1.VirtualClusterPolicyList
+		if err := client.List(ctx, &policies); err != nil {
+			return err
+		}
+
+		crd := &apiextensionsv1.CustomResourceDefinition{}
+		if err := client.Get(ctx, types.NamespacedName{Name: "virtualclusterpolicies.k3k.io"}, crd); err != nil {
+			return err
+		}
+
+		printerColumns := makePrinterColumns(crd)
+		table := makeTable(printerColumns)
+
+		for _, vcp := range policies.Items {
+			table.Rows = append(table.Rows, makeRow(&vcp, printerColumns))
+		}
+
+		printer := printers.NewTablePrinter(printers.PrintOptions{})
+
+		return printer.PrintObj(table, clx.App.Writer)
+	}
+}
+
+func makePrinterColumns(crd *apiextensionsv1.CustomResourceDefinition) []apiextensionsv1.CustomResourceColumnDefinition {
+	printerColumns := []apiextensionsv1.CustomResourceColumnDefinition{
+		{Name: "Name", Type: "string", Format: "name", Description: "Name of the Resource", JSONPath: ".metadata.name"},
+	}
+
+	for _, version := range crd.Spec.Versions {
+		if version.Name == "v1alpha1" {
+			printerColumns = append(printerColumns, version.AdditionalPrinterColumns...)
+			break
+		}
+	}
+
+	return printerColumns
+}
+
+func makeTable(printerColumns []apiextensionsv1.CustomResourceColumnDefinition) *metav1.Table {
+	var columnDefinitions []metav1.TableColumnDefinition
+
+	for _, col := range printerColumns {
+		columnDefinitions = append(columnDefinitions, metav1.TableColumnDefinition{
+			Name:        col.Name,
+			Type:        col.Type,
+			Format:      col.Format,
+			Description: col.Description,
+			Priority:    col.Priority,
+		})
+	}
+
+	return &metav1.Table{
+		TypeMeta:          metav1.TypeMeta{APIVersion: "meta.k8s.io/v1", Kind: "Table"},
+		ColumnDefinitions: columnDefinitions,
+	}
+}
+
+func makeRow(obj runtime.Object, printerColumns []apiextensionsv1.CustomResourceColumnDefinition) metav1.TableRow {
```

Comment:
nit: Since this is used by policy list and cluster list can we move it to separate file including makerow,makecells, etc.